### ISSUE TITLE
feat: Web API 

### ISSUE THREAD

**Issue by 0guanhua0** [Voyz/ibeam#227]
**Describe the feature**
[Web API](https://www.interactivebrokers.com/campus/ibkr-api-page/getting-started/#webapi)
[Client Portal Web API](https://www.interactivebrokers.com/campus/ibkr-api-page/getting-started/#legacy-webapi)

Interactive Brokers is merging our web-based API products into a single, comprehensive IBKR Web API, bringing the features of the Client Portal Web API, Digital Account Management, and the Flex Web Service together in a unified interface, accessible by a shared means of authorization and authentication: OAuth 2.0. 

**Possible implications**
update

**Comment by Voyz** [comment]
Hi @0guanhua0 please note that OAuth support is currently being implemented in IBind: https://github.com/Voyz/ibind/pull/34

This 'new' IBKR Web API still seems to be not production ready, seeing that its documentation is not completed.

Could you elaborate on what is the enhancement you're suggesting here?

**Comment by 0guanhua0** [comment]
thx for the info. let me try

sometimes ibeam lost connection. maybe due to ibkr maintenance. 
need to manually run [2fa](https://www.ibkrguides.com/securelogin/sls/twofactorauth.htm)

so maybe can get auto auth with new web api

**Comment by Voyz** [comment]
Yes, sadly the connection is sometimes lost and has to be remade, hopefully the new API will be more reliable. The moment it is available I'll look into it.

Is there anything we could discuss here that you'd like see implemented?

**Comment by 0guanhua0** [comment]
no thx

**Comment by hughandersen** [comment]
@0guanhua0 If you have access to IBKR OAuth then try out the new pull request oauth, I'm interested to hear your comments and I can help if needed.

**Comment by 0guanhua0** [comment]
> @0guanhua0 If you have access to IBKR OAuth then try out the new pull request oauth, I'm interested to hear your comments and I can help if needed.

let me try oauth

https://github.com/nsirons/ibkr_web_client?tab=readme-ov-file#setup

clean doc. ibkr doc is a mess

https://www.interactivebrokers.com/webtradingapi/oauth.pdf

```
1.1 Generation with openssl
A common method to generate the first three parameters is to use openssl:
openssl dhparam -outform PEM 2048 -out dhparam.pem
```

```
guanhua@arch ~ % openssl dhparam -outform PEM 2048 -out dhparam.pem
dhparam: Extra option: "2048"
dhparam: Use -help for summary.
```

ibkr doc cmd fail

```
guanhua@arch ~ % openssl dhparam -outform PEM -out dhparam.pem 2048
```

**Comment by Bomper** [comment]
> ibkr doc is a mess

Haha have you looked in the REST docs at how many endpoints return contract details?

- [Contract information by Contract ID](https://www.interactivebrokers.com/campus/ibkr-api-page/cpapi-v1/#info-conid-contract)
- [Search SecDef information by conid](https://www.interactivebrokers.com/campus/ibkr-api-page/cpapi-v1/#secdef-info-contract)
- [Search the security definition by Contract ID](https://www.interactivebrokers.com/campus/ibkr-api-page/cpapi-v1/#trsrv-conid-contract)

And how many names there are for the underlying conid field? `undConid`, `underlyingConid`, `underlying_con_id`, `underlying_conid`

**Comment by 0guanhua0** [comment]
> > ibkr doc is a mess
> 
> Haha have you looked in the REST docs at how many endpoints return contract details?
> 
>     * [Contract information by Contract ID](https://www.interactivebrokers.com/campus/ibkr-api-page/cpapi-v1/#info-conid-contract)
> 
>     * [Search SecDef information by conid](https://www.interactivebrokers.com/campus/ibkr-api-page/cpapi-v1/#secdef-info-contract)
> 
>     * [Search the security definition by Contract ID](https://www.interactivebrokers.com/campus/ibkr-api-page/cpapi-v1/#trsrv-conid-contract)
> 
> 
> And how many names there are for the underlying conid field? `undConid`, `underlyingConid`, `underlying_con_id`, `underlying_conid`

wow

big name like ibkr should provide clean product, no?

maybe no cash from api so no care
